models/GATStack.py

```diff
@@ -68,15 +71,71 @@ def __init__(
         )
         self.convs.append(conv)
         self.batch_norms.append(BatchNorm(self.hidden_dim))
+        self.__conv_node_features__()
         super()._multihead(
-            output_dim,
-            num_nodes,
-            output_type,
-            config_heads,
-            ilossweights_hyperp,
-            loss_weights,
-            ilossweights_nll,
+            num_nodes, ilossweights_hyperp, loss_weights, ilossweights_nll
         )
 
+    def __conv_node_features__(self):
```

Comment:
```suggestion
    def __conv_node_features(self):
```

Reply:
This kind of looks like a place to use CRTP (https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern), but I have to think more about it